contexts/global-effects.tsx

```diff
@@ -4,60 +4,80 @@ import {useSession} from "next-auth/react";
 import {useRouter} from "next/router";
 
 import {useAppState} from "contexts/app-state";
-
-import { CustomSession } from "interfaces/custom-session";
+import { changeLoadState } from "contexts/reducers/change-load";
 
 import {useAuthentication} from "x-hooks/use-authentication";
+import useChain from "x-hooks/use-chain";
 import {useDao} from "x-hooks/use-dao";
 import {useNetwork} from "x-hooks/use-network";
 import {useRepos} from "x-hooks/use-repos";
 import {useSettings} from "x-hooks/use-settings";
-import { useTransactions } from "x-hooks/use-transactions";
+import {useTransactions} from "x-hooks/use-transactions";
 
 const _context = {};
 
 export const GlobalEffectsContext = createContext(_context);
 export const GlobalEffectsProvider = ({children}) => {
 
-  const {state} = useAppState();
+  const {state, dispatch} = useAppState();
   const {query} = useRouter();
   const session = useSession();
 
   const dao = useDao();
   const repos = useRepos();
-  const auth = useAuthentication();
+  const { chain } = useChain();
   const network = useNetwork();
   const settings = useSettings();
+  const auth = useAuthentication();
   const transactions = useTransactions();
 
-  useEffect(dao.start, [state.Settings]);
-  useEffect(dao.changeNetwork, [state.Service?.active, state.Service?.network?.active?.networkAddress]);
-
-  useEffect(repos.loadRepos, [query?.network , state?.Service?.network?.lastVisited]);
-  useEffect(repos.updateActiveRepo, [query?.repoId, state.Service?.network?.repos]);
-
-  useEffect(auth.validateGhAndWallet,
-            [(session?.data as CustomSession),
-              state.currentUser?.walletAddress,
-              // asPath.includes('developers'),
-              // asPath.includes('bounty'),
-              // asPath.includes('profile'),
-            ]);
-  useEffect(auth.updateWalletAddress, [state.currentUser]);
-  useEffect(auth.listenToAccountsChanged, [state.Service]);
-  useEffect(auth.updateWalletBalance, [state.currentUser?.walletAddress, state?.Service?.active?.network]);
+  const { supportedChains, connectedChain, currentUser, Service } = state;
+
+  function updateLoadingState() {
+    dispatch(changeLoadState(Object.values(state.spinners).some(v => v)));
+  }
+
+  useEffect(updateLoadingState, [state?.spinners]);
```

Comment:
spinners were made so we can individual loading states happening, I'm sure this is safer -- but we should apply this condition on a "action button" case. 

It would be easier to just add a `spinnerKey` to the `Button` component and add a spinner inside, if `state.spinners[spinnerKey]` is true (and possibly, disable btn)

that way we stop _what we really want_ while the user can still do other stuff